ts/annotations/annotations.src.ts

```diff
@@ -0,0 +1,1608 @@
+            public getLabelsAndShapesOptions(
+                baseOptions: AnnotationsOptions,
+                newOptions: Partial<AnnotationsOptions>
+            ): AnnotationsOptions
+            public init(): void;
+            public initLabel(labelOptions: DeepPartial<AnnotationsLabelsOptions>, index: number): AnnotationLabelType;
+            public initShape(shapeOptions: DeepPartial<AnnotationsShapesOptions>, index: number): AnnotationShapeType;
+            public redraw(animation?: boolean): void;
+            public redrawItem(item: AnnotationControllable, animation?: boolean): void;
+            public redrawItems(items: Array<AnnotationControllable>, animation?: boolean): void;
+            public remove(): void;
+            public render(): void;
+            public renderItem(item: AnnotationControllable): void;
+            public setClipAxes(): void;
+            public setControlPointsVisibility(visibile?: boolean): void;
+            public setLabelCollector(): void;
+            public setOptions(userOptions: AnnotationsOptions): void;
+            public setVisibility(visible?: boolean): void;
+            public update(userOptions: DeepPartial<AnnotationsOptions>): void;
+        }
+        interface AnnotationChart extends Chart {
+            annotations: Array<Annotation>;
+            controlPointsGroup: SVGElement;
+            options: { annotations: Array<AnnotationsOptions> };
+            plotBoxClip: SVGElement;
+            addAnnotation(userOptions: AnnotationsOptions, redraw?: boolean): void;
+            drawAnnotations(): void;
+            initAnnotation(userOptions: AnnotationsOptions): Annotation;
+            removeAnnotation(idOrAnnotation: (number|string|Annotation)): void;
+        }
+        interface AnnotationControllableLabel {
+            itemType: 'label';
+        }
+        interface AnnotationControlPointEventsOptionsObject {
+            drag?: Function;
+        }
+        interface AnnotationControlPointOptionsObject {
+            draggable?: undefined;
+            events: AnnotationControlPointEventsOptionsObject;
+            height: number;
+            index?: number;
+            positioner: AnnotationControlPointPositioner;
+            style: CSSObject;
+            symbol: string;
+            visible: boolean;
+            width: number;
+        }
+        interface AnnotationControlPointPositioner {
+            (target: AnnotationControllable): PositionObject;
+        }
+        type AnnotationDraggableValue = (''|'x'|'y'|'xy');
+        type AnnotationLabelType = AnnotationControllableLabel;
+        type AnnotationShapeType = AnnotationControllable;
+        interface AnnotationMockPointOptionsObject {
+            x: number;
+            y: number;
+            xAxis?: (number|string|Axis|null);
+            yAxis?: (number|string|Axis|null);
+        }
+        interface AnnotationPoint extends Point {
+            series: AnnotationSeries;
```

Comment:
Ah, here is the definition for `AnnotationPoint`. One of other comments (about `mock`) doesn't matter, sorry.

Reply:
See my second comment.